jenkins-jobs/pipelines/connector_tests_pipeline.groovy

```diff
@@ -0,0 +1,151 @@
+                                string(name: 'BRANCH_CORE', value: params.BRANCH_CORE),
+                                string(name: 'SOURCE_URL', value: params.SOURCE_URL),
+                                booleanParam(name: 'PRODUCT_BUILD', value: params.PRODUCT_BUILD)
+                        ]
+                    }
+                }
+
+                stage('Invoke_sqlserver') {
+                    when {
+                        expression { params.SQLSERVER_TEST }
+                    }
+                    steps {
+                        build job: 'connector-debezium-sqlserver-matrix-test', parameters: [
+                                string(name: 'REPOSITORY_CORE', value: params.REPOSITORY_CORE),
+                                string(name: 'BRANCH_CORE', value: params.BRANCH_CORE),
+                                string(name: 'SOURCE_URL', value: params.SOURCE_URL),
+                                booleanParam(name: 'PRODUCT_BUILD', value: params.PRODUCT_BUILD)
+                        ]
+                    }
+                }
+            }
+        }
+
+        stage('Set_label') {
+            steps {
+                script {
+                    // check if all passed and store ran builds
+                    def child_builds = []
+                    def databases = ['db2', 'mongodb', 'mysql', 'oracle', 'postgresql', 'sqlserver']
+                    for (db in databases) {
+                        def run_test_arg = db.toUpperCase() + "_TEST"
+                        def job_name = "connector-debezium-" + db + "-matrix-test"
+                        if (!params[run_test_arg]) {
+                            continue
+                        }
+                        def build = jenkins.model.Jenkins.instance.getItem(job_name).lastBuild
+                        if(build.getResult() != hudson.model.Result.SUCCESS) {
+                            println "Not all ran tests passed, skipping label"
+                            return
+                        } else {
+                            child_builds << build
+                        }
+                    }
+                    if (!child_builds) {
+                        return
+                    }
+
+                    // determine labels for builds to set
+                    def label = params.LABEL
+                    def label_db2 = label
+                    if (!label && params.PRODUCT_BUILD) {
+                        def version_match = params.SOURCE_URL =~ /.*\/debezium-([^-]+)-src.zip$/
+                        if (version_match && version_match[0][1]) {
+                            def version = version_match[0][1].toString()
+                            label = version
+                            label_db2 = version
+                        } else {
+                            throw new IllegalArgumentException("Debezium version of product build couldn't be parsed from SOURCE_URL")
```

Comment:
No error needed (and if it were fail() is the right way to do it).

We either

- use the explicitly provided label if defined
- or use the branch name for upstream builds
- or parse the version for source URL for  downstream build
- or don't set the name if any of the above is possible

Reply:
this is for the scenario when PRODUCT_BUILD = true and version cannot be parsed from SOURCE_URL. I think in this case user should be notified, that this happended. This is basically scenario 4 but with error message. I could change this to fail() or at least println so user can see, that this happened.

Reply:
The label setting is just a convenience functionality and we definitively shouldn't mark the build as failure. Logging a message to stdout would be sufficient I guess. However I think the "error" is quite obviously observable by the fact that the trigger job build wont have its display name set :) . 

IMHO one of the cases where "error" management is not worth the extra complexity added to the pipeline script